### PR TITLE
Softening the Autolathe Nerf

### PR DESCRIPTION
**Pull request by Sly9sky**
## About The Pull Request
Autolathes will no longer print ammo magazines that are less than full. They also are able to have their print quality restored with very-high-quality manipulators installed.

## Why It's Good For The Game
Autolathes printing ammo magazines that are less than full did not entirely seem sensible to the point of nerfing their print quality, it did not make sense that less ammunition would be printed and cost the same amount of materials. 
I also desired to give another small way of restoring print quality to an autolathe such that players do not have to entirely depend on Lazarus to rejuvenate their printed product, especially when they are not present.

## Changelog
:cl:
tweak: Autolathes will no longer print magazines that are less than full
tweak: Autolathes can have their print quality restored with very-high-quality parts
/:cl:

### CODE REVIEW

**Comment by Sly9sky** [comment]
Currently changing my code to line up with the up-to-date code (mainly around qdel and content/stored_ammo), as I worked with the original autolathe nerf PR on Eris that came before optimizations.

**Review by EvilJackCarver — APPROVED** [review]

**Review by EvilJackCarver — CHANGES_REQUESTED** [review]
Hey, I've gotta revoke the approval, just double-checked something with mine and that reminded me I never checked to see if you flagged your edits.

Please flag your edits in a comment (e.g. `//Eclipse edit`) so we can see that we've edited it should it get overwritten by an upstream merge - if you don't, and someone else changes the same things, it'll get overwritten.

**Comment by Sly9sky** [comment]
> Hey, I've gotta revoke the approval, just double-checked something with mine and that reminded me I never checked to see if you flagged your edits.
> 
> Please flag your edits in a comment (e.g. `//Eclipse edit`) so we can see that we've edited it should it get overwritten by an upstream merge - if you don't, and someone else changes the same things, it'll get overwritten.
> 
> For more information, check [the Manual of Style](https://github.com/Eclipse-Station/NEV-Northern-Light/blob/master/CONTRIBUTING.md#editing-files-and-sprites-for-eclipse).

All Eclipse edits flagged.

File: code/modules/scrap/oldificator.dm [hunk 1]
```diff
@@ -145,13 +145,13 @@
 
 /obj/item/ammo_magazine/make_old(low_quality_oldification)
 	var/del_count = rand(0, stored_ammo.len)
-	if(low_quality_oldification)
+	if(!low_quality_oldification) // Eclipse edit
```
Comment (drexample) [inline code/modules/scrap/oldificator.dm:148]: Wait, why are you modifying oldificator?

Comment (Sly9sky) [reply code/modules/scrap/oldificator.dm:148]: This change ensures that autolathes don't print less-than-full ammo. It doesn't mess with the part of it that oldifies maintenance ammo though, those will still spawn as usually less-than-full

**Review by drexample — APPROVED** [review]

**Review by EvilJackCarver — COMMENTED** [review]
Once you fix what Nestor needs, I don't see any reason this can't be merged.

File: code/modules/scrap/oldificator.dm [hunk 2]
```diff
@@ -375,4 +375,4 @@
 			var/obj/item/trash_mod = new trash_mod_path
 			if(SEND_SIGNAL(trash_mod, COMSIG_IATTACK, src, null))
 				break
-			QDEL_NULL(trash_mod)
+			QDEL_NULL(trash_mod)
```
[the remark this answers is not in the document]
Comment (Sly9sky) [reply code/modules/scrap/oldificator.dm:378]: Gotcha, doing so now

**Comment by Sly9sky** [comment]
> Once you fix what Nestor needs, I don't see any reason this can't be merged.

Done

**Review by EvilJackCarver — APPROVED** [review]

**Pull request merged**